ci(coverage): codecov 'latest' endpoint race breaks SHA256 check

## Symptom

Merge run [24838802009](https://github.com/cynkra/blockyard/actions/runs/24838802009) failed in the `coverage` job during the "Upload coverage to Codecov" step:

```
sha256sum: WARNING: 1 computed checksum did NOT match
codecov: FAILED
##[error]Process completed with exit code 1.
```

This dequeued PR #300 even though the failure is unrelated to its changes (docs/installer only).

## Root cause hypothesis

The step does three sequential curl calls against `https://cli.codecov.io/latest/linux/`:

```sh
curl -fsSL https://cli.codecov.io/latest/linux/codecov         -o ./codecov
curl -fsSL https://cli.codecov.io/latest/linux/codecov.SHA256SUM -o ./codecov.SHA256SUM
sha256sum -c codecov.SHA256SUM
```

`latest/` is a moving pointer. If Codecov rolls a new CLI release between the two downloads, we end up with a binary from version N and a checksum from N+1 — sha256sum then rejects the mismatch. Nothing on our side is broken.

## Fix options

1. **Pin the codecov-cli version** (e.g. via a dedicated step env var or renovate-tracked constant) and drop the SHA check, or keep it but fetch both files from the same versioned path.
2. Use the official `codecov/codecov-action@v5` GitHub Action — it handles pinning and integrity internally.

(1) is minimal; (2) follows upstream recommendations and gets us free updates.

## Impact

Flaky on a tight schedule; blocks merges whenever Codecov rolls their CLI.

--- a/.github/workflows/merge.yml
+++ b/.github/workflows/merge.yml
@@ -320,9 +320,13 @@ jobs:
       - name: Upload coverage to Codecov
         env:
           CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+          # Pin codecov-cli to avoid races on the `latest/` endpoint,
+          # where a release rolling between the two curl calls left us
+          # with a binary and checksum from different versions.
+          CODECOV_CLI_VERSION: v11.2.8
         run: |
-          curl -fsSL https://cli.codecov.io/latest/linux/codecov -o ./codecov
-          curl -fsSL https://cli.codecov.io/latest/linux/codecov.SHA256SUM -o ./codecov.SHA256SUM
+          curl -fsSL "https://cli.codecov.io/${CODECOV_CLI_VERSION}/linux/codecov" -o ./codecov
+          curl -fsSL "https://cli.codecov.io/${CODECOV_CLI_VERSION}/linux/codecov.SHA256SUM" -o ./codecov.SHA256SUM
           sha256sum -c codecov.SHA256SUM
           chmod +x ./codecov
           for f in coverage-*.out; do

